bayesian/utils.py

```diff
@@ -410,87 +411,123 @@ class CveByDateEcosystemUtils:
     g.V().has('modified_date', modified_date)\
-    .has('vertex_label','CVE')\
-    .has('ecosystem',ecosystem).as('cve')\
-    .coalesce(\
-    inE('has_cve').outV().as('epv').select('cve','epv').by(valueMap()),\
-    select('cve').by(valueMap()))\
+    .has('cecosystem',ecosystem)\
+    .has('cve_sources',cve_sources)\
+    .valueMap()\
     """
 
-    def __init__(self, bydate, ecosystem=None):
-        """Constructor."""
-        self._bydate = bydate
-        self._ecosystem = ecosystem
+    cve_nodes_by_date_ecosystem_script_template_all = """\
+    g.V().has('modified_date', modified_date)\
+    .has('cecosystem',ecosystem)\
+    .valueMap()\
+    """
 
-    def get_cves_by_date(self):
-        """Call graph and get CVEs by date."""
-        script = self.cve_nodes_by_date_script_template
-        bindings = {'modified_date': self._bydate}
-        return self.get_cves(script, bindings)
+    def __init__(self, result_type, cve_id, cve_sources='all', bydate=None,
+                 ecosystem='all', date_range=7):
+        """Constructor."""
+        self._result_type = result_type
+        self._cve_id = cve_id
+        if cve_id is None:
+            self._bydate = bydate
+            self._ecosystem = ecosystem
+            self._cve_sources = cve_sources
+            self._date_range = date_range
 
     def get_cves_by_date_ecosystem(self):
         """Call graph and get CVEs by date and ecosystem."""
-        script = self.cve_nodes_by_date_ecosystem_script_template
-        self._ecosystem = self._ecosystem.lower()
-        bindings = {'modified_date': self._bydate, 'ecosystem': self._ecosystem}
-        return self.get_cves(script, bindings)
+        didx = pd.date_range(end=self._bydate, periods=self._date_range)
+        gremlin_json = []
 
-    def get_cves(self, script, bindings):
-        """Call Gremlin and get the CVE information."""
-        json_payload = self.prepare_payload(script, bindings)
-        response = post(gremlin_url, json=json_payload)
-        cve_list = self.prepare_response(response.json())
+        try:
+            for dt in didx:
+                dt = datetime.datetime.strftime(dt, '%Y%m%d')
+                # Gremlin script execution
+                script = self.cve_nodes_by_date_ecosystem_script_template_all \
+                    if self._cve_sources == 'all' \
+                    else self.cve_nodes_by_date_ecosystem_script_template
+
+                self._ecosystem = self._ecosystem.lower()
+                bindings = {
+                    'modified_date': dt, 'ecosystem': self._ecosystem,
+                    'cve_sources': self._cve_sources
+                }
+                """Call Gremlin and get the CVE information."""
+                json_payload = self.prepare_payload(script, bindings)
+
+                response = post(gremlin_url, json=json_payload)
+                gremlin_json.extend(response.json().get('result', {}).get('data', []))
+        except Exception as e:
```

Comment:
return a simple excpetion. Also raise the exception to be caught in the cve_by_date endpoint

Reply:
You need to raise this exception to be caught in rest_api file